src/components/menu/menu.svelte

```diff
@@ -0,0 +1,235 @@
+<script context="module" lang="ts">
+  import type { HTMLAttributes } from 'svelte/elements'
+  declare global {
+    namespace JSX {
+      interface IntrinsicElements {
+        'leo-menu-item': HTMLAttributes<HTMLElement> & {
+          // Note: This should line up with Reacts key type, but we don't want
+          // to depend on React in this layer, so we just define it manually.
+          key?: string | number | null
+
+          // value?: string
+          children?: any
+        }
+      }
+    }
+  }
+</script>
+
+<script lang="ts">
+  import clickOutside from '../../svelteDirectives/clickOutside'
+
+  let isOpen = false
+  let popup: HTMLDivElement
+  let button: HTMLButtonElement
+
+  const slots = $$slots
+
+  $: menuItems = Array.from(
+    (
+      popup?.querySelector('.leo-menu-popup slot') as HTMLSlotElement
+    )?.assignedElements() ??
+      popup?.querySelectorAll('.leo-menu-popup > *') ??
+      []
+  )
+
+  $: {
```

Comment:
It worries me a bit that we're duplicating all this logic - the code is a bit finicky because it has to consider a bunch of different things, and I can pretty easily see a scenario where we fix a bug in one of these but not the other. It'd be nice if we could factor some of this out

Reply:
We won't be duplicating it anymore. I'll use the Menu component after you merge the floating-ui PR.